go_lib/hooks/internal_tls/hook.go

```diff
@@ -31,6 +31,9 @@ import (
 type GenSelfSignedTLSHookConf struct {
 	// SANs - list of domains to include into certificate
 	SANs []string
+	// SANsGenerator dynamic SANs generator based in hook input (e.x.: values like cluster domain)
+	SANsGenerator func(input *go_hook.HookInput) []string
```

Comment:
For certificates, signed by Kubernetes, we used a diffirent logic for the same purpose:

https://github.com/deckhouse/deckhouse/blob/c59c3dd101035b233ad35f28bc3f26eabcc11168/modules/150-user-authn/hooks/order_dex_tls_certificate.go#L31-L36

This logic should be either applied to this hook or changed according to new standards. I personally do not like dynamic san creation from any value, but using cluster and public domains is ok.

Reply:
I think we even want to go further and use the same structure in all modules. It sounds like an idea for significant refactoring.

Reply:
another PR I guess